### ISSUE TITLE
BUG: No definition provided for `no-redundant-comparison`, `prefer-number-is-safe-integer`, and `prefer-temporal`

### ISSUE THREAD

**Issue by ShayanTheNerd** [sindresorhus/eslint-plugin-unicorn#3141]
ESLint version: 10.4.1
eslint-plugin-unicorn version: 65.0.1

_eslint.config.ts_:
```ts title="eslint.config.ts"
import { defineConfig } from 'eslint/config';
import eslintPluginUnicorn from 'eslint-plugin-unicorn';

export default defineConfig([{
  name: 'project/unicorn',
  files: ['**/*.?([mc])[jt]s?(x)'],
  plugins: {
    unicorn: eslintPluginUnicorn,
  },
  rules: {
    'unicorn/no-redundant-comparison': 'warn',
    'unicorn/prefer-number-is-safe-integer': 'warn',
    'unicorn/prefer-temporal': 'warn',
  },
}]);
```

✅ Expected behavior: no issues.
❌ Actual behavior: After running `pnpm eslint`, I get a similar error for either of these three rules:
> Oops! Something went wrong! :(
>
> ESLint: 10.4.1
>
> TypeError: Key "rules": Key "unicorn/no-redundant-comparison": Could not find "prefer-temporal" in plugin "unicorn".
> at `throwRuleNotFoundError` (_project-path/node_modules/.pnpm/eslint@10.4.1_jiti@2.7.0/node_modules/eslint/lib/config/config.js:100:16_)

**Comment by sindresorhus** [comment]
They are not released yet. Please try a little bit harder before opening an issue.

**Comment by ShayanTheNerd** [comment]
That's okay, but how was I supposed to figure out they weren't released yet? From a user's perspective, there are several strong signals that normally imply a rule is available:
- The rule has a dedicated documentation page.
- The rule appears in [the rules table](https://github.com/sindresorhus/eslint-plugin-unicorn#rules) and isn't flagged as "upcoming", "unreleased", etc.
- The rule name and implementation are discoverable through repository/codebase navigation.
- In most projects, the documentation is kept in sync with the releases, or unreleased features are clearly marked.

I'm not saying the rules shouldn't be documented before release; only that there doesn't seem to be an obvious way for users to distinguish between released and unreleased rules when browsing the repository.
